{
    "url": "https://api.github.com/repos/spack/spack/issues/2687",
    "repository_url": "https://api.github.com/repos/spack/spack",
    "labels_url": "https://api.github.com/repos/spack/spack/issues/2687/labels{/name}",
    "comments_url": "https://api.github.com/repos/spack/spack/issues/2687/comments",
    "events_url": "https://api.github.com/repos/spack/spack/issues/2687/events",
    "html_url": "https://github.com/spack/spack/pull/2687",
    "id": 198179274,
    "node_id": "MDExOlB1bGxSZXF1ZXN0OTk3NzMwNTM=",
    "number": 2687,
    "title": "Fix Python issue with build failures; Add test case for build failures.",
    "user": {
        "login": "tgamblin",
        "id": 299842,
        "node_id": "MDQ6VXNlcjI5OTg0Mg==",
        "avatar_url": "https://avatars.githubusercontent.com/u/299842?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/tgamblin",
        "html_url": "https://github.com/tgamblin",
        "followers_url": "https://api.github.com/users/tgamblin/followers",
        "following_url": "https://api.github.com/users/tgamblin/following{/other_user}",
        "gists_url": "https://api.github.com/users/tgamblin/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/tgamblin/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/tgamblin/subscriptions",
        "organizations_url": "https://api.github.com/users/tgamblin/orgs",
        "repos_url": "https://api.github.com/users/tgamblin/repos",
        "events_url": "https://api.github.com/users/tgamblin/events{/privacy}",
        "received_events_url": "https://api.github.com/users/tgamblin/received_events",
        "type": "User",
        "site_admin": false
    },
    "labels": [
        {
            "id": 73908754,
            "node_id": "MDU6TGFiZWw3MzkwODc1NA==",
            "url": "https://api.github.com/repos/spack/spack/labels/bug",
            "name": "bug",
            "color": "fc2929",
            "default": true,
            "description": null
        },
        {
            "id": 446645732,
            "node_id": "MDU6TGFiZWw0NDY2NDU3MzI=",
            "url": "https://api.github.com/repos/spack/spack/labels/build-environment",
            "name": "build-environment",
            "color": "bfdadc",
            "default": false,
            "description": null
        }
    ],
    "state": "closed",
    "locked": false,
    "assignee": null,
    "assignees": [],
    "milestone": null,
    "comments": 0,
    "created_at": "2016-12-30T18:24:19Z",
    "updated_at": "2016-12-30T19:37:51Z",
    "closed_at": "2016-12-30T19:37:51Z",
    "author_association": "MEMBER",
    "active_lock_reason": null,
    "pull_request": {
        "url": "https://api.github.com/repos/spack/spack/pulls/2687",
        "html_url": "https://github.com/spack/spack/pull/2687",
        "diff_url": "https://github.com/spack/spack/pull/2687.diff",
        "patch_url": "https://github.com/spack/spack/pull/2687.patch"
    },
    "body": "Fixes issue reported by @schulzm.  Fixes #2688.\r\n\r\n```\r\n==> Executing phase : 'install'\r\nProcess Process-1:\r\nTraceback (most recent call last):\r\n  File \"/usr/apps/python/lib/python2.7/multiprocessing/process.py\", line 258, in _bootstrap\r\n    self.run()\r\n  File \"/usr/apps/python/lib/python2.7/multiprocessing/process.py\", line 114, in run\r\n    self._target(*self._args, **self._kwargs)\r\n  File \"/g/g23/schulz/projects/spack/lib/spack/spack/build_environment.py\", line 564, in child_execution\r\n    package_context = get_package_context(tb)\r\n  File \"/g/g23/schulz/projects/spack/lib/spack/spack/build_environment.py\", line 627, in get_package_context\r\n    if isinstance(obj, spack.PackageBase):\r\nAttributeError: 'module' object has no attribute 'PackageBase'\r\n```\r\n\r\n- [x] Fix bug from imports cleaned up in #2681.\r\n- [x] Add this issue to the regression tests with a new mock package that raises an `InstallError`.\r\n\r\n@schulzm: can you please verify?",
    "performed_via_github_app": null
}